projects/ccf-rui/src/app/modules/left-sidebar/left-sidebar.component.ts

```diff
@@ -72,4 +78,24 @@ export class LeftSidebarComponent {
   setSideFromLabel(label: 'L' | 'R'): void {
     this.model.setSide(label === 'L' ? 'left' : 'right');
   }
+
+  /**
+   * Handles toggling previous registration blocks visibility.
+   * When making them visible, it updates current structures to 20%
+   * opacity; when making not visible it sets them back to their
+   * previous opacity.
+   * @param visible the state to set the visibility to.
+   */
+  togglePreviousRegistrationBlocks(visible: boolean): void {
+    this.model.setShowPrevious(visible);
+    if (!visible) {
+      this.model.setAnatomicalStructures(this.previousVisibilityItems);
+    } else {
+      this.previousVisibilityItems = [...this.model.snapshot.anatomicalStructures] as VisibilityItem[];
+      const newAnotomicalStructures = this.previousVisibilityItems.map(structure => {
+        return { ...structure, opacity: 20 };
```

Comment:
Max opacity should be 20%. If the item is lower than 20% already, then keep it at that value.